// embla-carousel-auto-height/embla-carousel-auto-height.esm.js
const defaultOptions = {
  active: true,
  breakpoints: {},
  destroyHeight: 'auto'
};

function AutoHeight(userOptions = {}) {
  let options;
  let emblaApi;
  let slideBounds = [];
  let slideHeights = [];
  const heightEvents = ['select', 'pointerUp'];
  const inViewThreshold = 0.5;
  function init(emblaApiInstance, optionsHandler) {
    emblaApi = emblaApiInstance;
    const {
      mergeOptions,
      optionsAtMedia
    } = optionsHandler;
    const optionsBase = mergeOptions(defaultOptions, AutoHeight.globalOptions);
    const allOptions = mergeOptions(optionsBase, userOptions);
    options = optionsAtMedia(allOptions);
    const {
      options: {
        axis
      },
      slidesInView,
      slideRects
    } = emblaApi.internalEngine();
    if (axis === 'y') return;
    slideBounds = slidesInView.findSlideBounds(undefined, inViewThreshold);
    slideHeights = slideRects.map(rect => rect.height);
    heightEvents.forEach(evt => emblaApi.on(evt, setContainerHeight));
    setContainerHeight();
  }
  function destroy() {
    heightEvents.forEach(evt => emblaApi.off(evt, setContainerHeight));
    setContainerHeight(undefined, 'destroy');
  }
  function highestInView() {
    const {
      slidesInView,
      target
    } = emblaApi.internalEngine();
    const inViewIndexes = slidesInView.check(target.get(), slideBounds);
    const heights = inViewIndexes.map(index => slideHeights[index]);
    return heights.reduce((a, b) => Math.max(a, b), 0);
  }
  function setContainerHeight(_, evt) {
    const height = evt === 'destroy' ? options.destroyHeight : `${highestInView()}px`;
    emblaApi.containerNode().style.height = height;
  }
  const self = {
    name: 'autoHeight',
    options: userOptions,
    init,
    destroy
  };
  return self;
}
AutoHeight.globalOptions = undefined;

export { AutoHeight as default };
